src/Apps/Order/Routes/Shipping/useAddressAutocomplete.ts

```diff
@@ -0,0 +1,135 @@
+console.log({ smartyCreds })
+const key = smartyCreds.key
+
+export type SuggestionJSON = {
+  city: string
+  entries: number
+  secondary: string
+  state: string
+  street_line: string
+  zipcode: string
+}
+
+export interface AddressSuggestion extends AutocompleteInputOptionType {
+  address: Omit<Address, "name">
+  entries: number
+}
+
+export const useAddressAutocomplete = () => {
+  const [result, setResult] = useState<SuggestionJSON[]>([])
+
+  // Add feature flag checks including country, etc here later
+  // console.log({ key, kind: typeof key })
+  const enabled = !!key
+
+  const fetchSuggestions = useCallback(
+    async (searchParam: string, selectedParam?: string) => {
+      console.log({ key })
+      if (!key) return null
+      let url = `https://us-autocomplete-pro.api.smarty.com/lookup?key=${encodeURIComponent(
+        key
+      )}&prefer_ratio=3&search=${encodeURIComponent(searchParam)}`
+
+      if (selectedParam) {
+        url += `&selected=${encodeURIComponent(selectedParam)}`
+      }
+
+      console.log({ url })
+      const response = await fetch(url, {
+        headers: {
+          Host: "us-autocomplete-pro.api.smartystreets.com",
+        },
+      })
+      console.log({ response })
+      const json = await response.json()
+      return json
+    },
+    []
+  )
+
+  const fetchForAutocomplete = useCallback(
+    // these are the parameters to the Smarty API call
+    async (searchParam: string, selectedParam?: string) => {
+      if (searchParam.length < 5) {
+        console.log("type more...")
+        setResult([])
+        return
+      }
+
+      try {
+        const result = await fetchSuggestions(searchParam, selectedParam)
+
+        console.log({ result })
+        setResult(result.suggestions)
+      } catch (e) {
+        console.error(e)
+      }
+    },
+    [fetchSuggestions]
+  )
+
+  // debounce requests
+  const debouncedFetchForAutocomplete = useDebounce({
+    // high debounce for testing to not eat up our credits
+    delay: 700,
+    callback: fetchForAutocomplete,
+  })
+
+  const buildAddressText = (suggestion): string => {
+    let whiteSpace = ""
+    let secondaryExtraInformation = ""
+    if (suggestion.secondary) {
+      if (suggestion.entries > 1) {
+        secondaryExtraInformation = " (" + suggestion.entries + " entries)"
+      }
+      whiteSpace = " "
+    }
+    return (
+      suggestion.street_line +
+      whiteSpace +
+      suggestion.secondary +
+      secondaryExtraInformation +
+      " " +
+      suggestion.city +
+      ", " +
+      suggestion.state +
+      " " +
+      suggestion.zipcode
+    )
+  }
+
+  const autocompleteOptions: Array<AddressSuggestion> = result.map(
+    suggestion => {
+      const text = buildAddressText(suggestion)
+      return {
+        text,
+        value: text,
+        entries: suggestion.entries,
+        address: {
+          addressLine1: suggestion.street_line,
+          addressLine2: suggestion.secondary,
```

Comment:
We observed that `secondary` should actually be appended to `street_line` as is common with things like apartment numbers.

Reply:
The logic here could be shared with the logic in our buildAddressText code.

Reply:
Maybe a followup for later - currently the line 2 input specifically says apt number even though it is not.

OTOH, doing this right would help us avoid the address verification modal if we ignore the zipcode +4 for VERIFIED_NO_CHANGES results in the future.

Probably a followup for design since this line 2 label/placeholder text is very standard, though wrong, but the proposal would be:
- stick secondary info on line 1 where the verification api expects it
- update verification api to consider a zip code that is correct but lacking the zip+4 as still verified without changes
- force receives result and proceeds without showing the user a modal.